embassy-rp/src/timer.rs

```diff
@@ -68,9 +68,16 @@ impl Driver for TimerDriver {
         })
     }
 
-    fn set_alarm(&self, alarm: AlarmHandle, timestamp: u64) {
+    fn set_alarm(&self, alarm: AlarmHandle, timestamp: u64) -> bool {
         let n = alarm.id() as usize;
         critical_section::with(|cs| {
+            let now = self.now();
```

Comment:
I think changing the order here is not correct. If you check `timestamp <= t` first, and set it after, `timestamp` might have elapsed between both, causing the alarm to never fire.

Reply:
I've tried to address your feedback for all the three drivers. 

They are now doing almost exactly what they did before my changes (checks for timestamp-in-the-past are restored to be at their original places in the code and they are disarming the alarm as well. Only difference now is they return `false` rather than calling the callback synchronously). Code is a bit more verbose as disarming the alarm is not happening as a side effect of calling `trigger_alarm` but we are talking a line or two extra, so I hope this is acceptable.